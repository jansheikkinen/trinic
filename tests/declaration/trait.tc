// Trait

trait TestTrait
  function testFunction(void) -> void
  function testFunctionB(void) -> void
end
