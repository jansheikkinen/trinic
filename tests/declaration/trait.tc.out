
### LEXER ###
000:        TRAIT,     (null), (003, 005)
001:   IDENTIFIER,  TestTrait, (003, 015)
002:     FUNCTION,     (null), (004, 010)
003:   IDENTIFIER, testFunction, (004, 023)
004:            (,     (null), (004, 024)
005:         VOID,     (null), (004, 028)
006:            ),     (null), (004, 029)
007:           ->,     (null), (004, 031)
008:         VOID,     (null), (004, 036)
009:     FUNCTION,     (null), (005, 010)
010:   IDENTIFIER, testFunctionB, (005, 024)
011:            (,     (null), (005, 025)
012:         VOID,     (null), (005, 029)
013:            ),     (null), (005, 030)
014:           ->,     (null), (005, 032)
015:         VOID,     (null), (005, 037)
016:          END,     (null), (006, 003)

### AST GENERATION ###
Declaration TRAIT TestTrait
  Declaration FUNCTION  testFunction
  Declaration FUNCTION  testFunctionB
